plugins/repository-gcs/src/main/java/org/elasticsearch/common/blobstore/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -189,14 +189,17 @@ boolean blobExists(String blobName) throws IOException {
      * @return an InputStream
      */
     InputStream readBlob(String blobName) throws IOException {
+        if (!blobExists(blobName)) {
+            throw new NoSuchFileException("Blob [" + blobName + "] does not exist");
+        }
```

Comment:
Ditto for GCS as I said for Azure <a href="https://github.com/elastic/elasticsearch/pull/19421/files#r71458452">here</a>.